app/controllers/concerns/unconfirmed_user_concern.rb

```diff
@@ -25,11 +25,23 @@ def track_user_already_confirmed_event
       errors: { email: [t('errors.messages.already_confirmed')] },
       user_id: @user.uuid,
     )
+    irs_attempts_api_tracker.user_registration_email_confirmation(
+      email: @email_address.email,
+      success: false,
+      errors: { email: [t('errors.messages.already_confirmed')] },
+      error_details: { email: [t('errors.messages.already_confirmed')] },
+    )
   end
 
   def stop_if_invalid_token
     result = email_confirmation_token_validator.submit
     analytics.user_registration_email_confirmation(**result.to_h)
+    irs_attempts_api_tracker.user_registration_email_confirmation(
+      email: @email_address&.email,
+      success: result.success?,
+      errors: result.errors,
+      error_details: result.to_h[:error_details],
```

Comment:
`errors` and `error_details` are pretty much the same information, just formatted differently. `error_details` is the newer one, should we just use that everywhere?

Reply:
This one's a little trickier since we don't know what the error might be. Assuming `result.errors` an empty array if there are no errors, maybe we could do something like:

```suggestion
      failure_reason: result.errors.join(', '),
```

Although, 🤔 this will result in including an empty-string `failure_reason` on success. It's probably cleaner to only send the client a `failure_reason` attribute if there's an error.

But maybe this is fine, and we modify (not necessarily in this PR) [the track_event method](https://github.com/18F/identity-idp/blob/main/app/services/irs_attempts_api/tracker.rb#L10-L22) to pull out `failure_reason` if it's not specified?

Reply:
Although this also makes me wonder if our schema would be better if `failure_reason` was actually an array (and renamed to `failure_reasons`).

I think it will be very rare that we have multiple failure reasons for an event, but then there are cases like this when we aren't necessarily handling something specific but something could have gone wrong.

I _think_ we're good to keep it as a `failure_reason` string, but 🤔...